document/AnnexE-Determinism

```diff
@@ -0,0 +1,136 @@
+# Annex E (informative) Producing deterministic SARIF log files { - #Annex-E }
+
+## General { #Annex-E-General }
+
+In certain circumstances, it is desirable for an analysis tool to produce deterministic output;
+that is, for it to produce identical output when run repeatedly over identical inputs.
+
+Certain build systems provide an example of when this is desirable.
+Consider a build system that caches the results of each build step.
+If the build is rerun, and the inputs to the step are identical
+(which the build system might determine, for example, by comparing timestamps,
+or by computing a hash of the inputs to the step and storing it along with the output from the step),
+then the build system can save time by not re-running the step, and simply
+using the existing outputs.
+
+In the case of SARIF, one could imagine a sequence of build steps where Steps A
+B, and C each run an analysis tool on a different set of targets, producing log files
+A.sarif, B.sarif, and C.sarif, and then build Step D performs an analysis on the aggregate
+of those log files. If the targets analyzed in Step B change
+but the targets analyzed in steps A and C do not, and if the contents of the
+SARIF log file are deterministic, then when the build is re-run, only Steps B and D
+need be performed.
+
+Authors of analysis tools are encouraged to provide a mechanism
+(for example, a command line option such as `--deterministic`) which instructs the tool to
+produce deterministic output.
+
+There are three issues to consider when producing deterministic output:
```

Comment:
If you are building a comprehensive list, we should also discuss two others. First, the paths issues, where there are deterministic relative paths against a non-deterministic root. Second, we may need some special language around baselining. Baselines are designed for run-over-run analysis, and so we can't toss 'baselineState' out. In other words, baselineState is designed to drive a specific diffing scenario. In order for it that value to be deterministic, results mgmt. tools need to rewrite baselines before check-in (so that all 'new' issues are 'existing') or the system needs a sufficiently subtle diff in order to accommodate transition from 'new' to 'existing'.